pkg/provider/bao/bao.go

```diff
@@ -0,0 +1,174 @@
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bao
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/signal"
+	"strings"
+
+	injector "github.com/bank-vaults/internal/pkg/baoinjector"
+	bao "github.com/bank-vaults/vault-sdk/vault"
+
+	"github.com/bank-vaults/secret-init/pkg/common"
+	"github.com/bank-vaults/secret-init/pkg/provider"
+)
+
+const (
+	ProviderName = "bao"
+	Re           = `(bao:)(.*)#(.*)`
+)
```

Comment:
Lets change these into vars (do the same for other providers) so that regexp expression issues can be caught on compile time rather than runtime
```suggestion
var (
	ProviderName = "bao"
	ProviderEnvRegex = regexp.MustCompile(`(bao:)(.*)#(.*)`)
)
```